microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,122 @@
+from lisa.sut_orchestrator.azure.tools import LisDriver
+from lisa.testsuite import simple_requirement
+from lisa.tools import Lsmod
+from lisa.util import LisaException, SkippedException
+from lisa.util.perf_timer import create_timer
+
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+    tags=[],
+)
+class gpu(TestSuite):
+    grid_skus = ["Standard_NV"]
+    cuda_skus = ["Standard_NC", "Standard_ND"]
+
+    def __init__(self, node: Node) -> None:
+        self.gpu_feature = node.features[Gpu]
+        self.serial_console = node.features[SerialConsole]
+
+    def _validate_node_reboot(
+        self, node: Node, case_name: str, case_path: Path
+    ) -> None:
+        try:
+            timer = create_timer()
+            node.reboot()
+            self.log.info(f"node '{node.name}' rebooted in {timer}")
+        except Exception as identifier:
+            # if there is any panic, fail before partial pass
+            self.serial_console.check_panic(saved_path=case_path, stage="reboot")
+
+            # if node cannot be connected after reboot, it should be failed.
+            if isinstance(identifier, LisaException) and str(identifier).startswith(
+                "cannot connect to TCP port"
+            ):
+                raise LisaException(f"after reboot, {identifier}")
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu.
+            TODO: validate the test params against GPU support test matrix
+                    when test matrix is available.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu],
+        ),
+    )
+    def validate_gpu_support(self, case_name: str, node: Node) -> None:
+        self.gpu_feature.check_support()
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies if gpu drivers are loaded fine.
+
+            Steps:
+            1. Validate if the VM SKU is supported for GPU.
+            2. Install LIS drivers if not already installed for Fedora and its
+                derived distros. Reboot the node
+            3. Install required gpu drivers on the VM and reboot the node. Validate gpu
+                drivers can be loaded successfully.
+
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu, SerialConsole],
+        ),
+    )
+    def validate_load_driver(self, case_name: str, node: Node) -> None:
+        case_path = self._create_case_log_path(case_name)
+
+        if isinstance(node.os, Redhat):
+            # install LIS driver if not already installed.
+            lisdriver = node.tools[LisDriver]
+            if not lisdriver.get_version():
+                lisdriver.install_driver()
+                self._validate_node_reboot(node, case_name, case_path)
+
+        node_runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
```

Comment:
Don't use vm_size in test case, it makes the test case can run Azure only. Write a method in feature to detect GRID or CUDA.

Reply:
I can move the logic to feature but detecting which one to install - GRID or CUDA depends on the VM size itself.

Reply:
It's ok, that's why the feature is needed. Make sure the logic is not in common part of feature, it should be in subclass of Azure.

Reply:
Done